finance/views.py

```diff
@@ -27,13 +27,28 @@ def get_context_data(self, *args, **kwargs):
         )
 
 
-class GetFinanceCMS(
+class DeprecatedGetFinance(
+    mixins.GetCMSPageMixin, PiTrackerContextData, TemplateView
+):
+    template_name = 'finance/get_finance_deprecated.html'
+    slug = EXPORT_READINESS_GET_FINANCE_SLUG + '-deprecated'
+
+
+class GetFinance(
     mixins.GetCMSPageMixin, PiTrackerContextData, TemplateView
 ):
     template_name = 'finance/get_finance.html'
     slug = EXPORT_READINESS_GET_FINANCE_SLUG
 
 
+class GetFinanceNegotiator(TemplateView):
+    def __new__(cls, *args, **kwargs):
+        if settings.FEATURE_FLAGS['UKEF_LEAD_GENERATION_ON']:
+            return GetFinance(*args, **kwargs)
+        else:
+            return DeprecatedGetFinance(*args, **kwargs)
```

Comment:
Not a fan of the name, we should have deprecated code around, but this is actually still in use depending on the flag... Not a big deal.

Reply:
marking code as deprecated is a step away from deleting it e.g, the deprecation warnings in Django when using the old style middleware.

`DeprecatedGetFinance` is used when the feature flag is turned off, and `DeprecatedGetFinance` will be deleted once the feature is to be turned on always.